src/Data/Algorand/Transaction.hs

```diff
@@ -106,7 +107,29 @@ data TransactionType
   | AssetFreezeTransaction
     {}
   deriving (Eq, Generic, Show)
-$(deriveJSON algorandTrainOptions 'PaymentTransaction) -- HERE!
+
+instance ToJSON TransactionType where
+  toJSON txType = toJSON (value :: Text)
+    where
+      value = case txType of
+        PaymentTransaction{} -> "pay"
+        ApplicationCallTransaction{} -> "appl"
+        AssetTransferTransaction{} -> "axfer"
+        KeyRegistrationTransaction -> "keyreg"
+        AssetConfigTransaction -> "acfg"
+        AssetFreezeTransaction -> "afrz"
+
+instance FromJSON TransactionType where
+  parseJSON o = do
+    value :: Text <- parseJSON o
+    pure $ case value of
+      "pay" -> PaymentTransaction{}
+      "appl" -> ApplicationCallTransaction{}
+      "axfer" -> AssetTransferTransaction{}
+      "keyreg" -> KeyRegistrationTransaction
+      "acfg" -> AssetConfigTransaction
+      "afrz" -> AssetFreezeTransaction
+      x -> error . T.unpack $ "Unmapped transaction type field name: " <> x
```

Comment:
Please use `fail` in `parseJSON`.

Reply:
ops.. I sure I've changed this copy paste to `fail`, maybe `cmd + z`. I'll do it